### ISSUE TITLE
Python Robot Name exercise test file

### ISSUE THREAD

**Issue by floydbj** [exercism/exercism#5240]
I may be wrong, but it seems there's a mistake in the test file for the Robot Name exercise on the Python track that is preventing valid solutions from passing.

The error I'm getting on the final test indicates the following:

self.assertRegex(name2, self.name_re)
E       AssertionError: Regex didn't match: '^[A-Z]{2}\\\d{3}$' not found in 'FV566CP146'

I believe the regular expression here is written incorrectly, and should instead be:

^[A-Z]{2}\d{3}$

In other words, there should be only one \ character before d

**Comment by glennj** [comment]
Why are there two robot names in that string? ("FV566CP146") Perhaps that's the problem.

**Comment by floydbj** [comment]
> Why are there two robot names in that string? ("FV566CP146") Perhaps that's the problem.

Honestly, I didn't even notice that when I posted. I have all of two months' programming experience so forgive me if I overlook something obvious.

That said, I still feel that something is going wrong. The following is the full error message:

`self = <robot_name_test.RobotNameTest testMethod=test_reset_name>

    def test_reset_name(self):
        # Set a seed
        seed = "Totally random."

        # Initialize RNG using the seed
        random.seed(seed)

        # Call the generator
        robot = Robot()
        name = robot.name

        # Reinitialize RNG using seed
        random.seed(seed)

        # Call the generator again
        robot.reset()
        name2 = robot.name
        self.assertNotEqual(name, name2)
>       self.assertRegex(name2, self.name_re)
E       AssertionError: Regex didn't match: '^[A-Z]{2}\\d{3}$' not found in 'FV566CP146'

robot_name_test.py:47: AssertionError`

So the test is simply running the generator, saving the robot name as 'name', then running reset and saving the new name as 'name2'. It checks to make sure that the two are not equal, then checks to make sure that 'name2' matches the expected format via comparison to the regular expression.

I've done manual testing of my methods via the following code:

`r1 = Robot()
print(r1.name)
r1.reset()
print(r1.name)`

After the reset, I get a new name for 'r1' in the correct format. I'm not seeing a string combining two names into one.

Furthermore, even if my methods weren't working properly, shouldn't I get a NEW string combining two random names every time I run the test? No matter how many times I run it, I always see the same thing, 'FV566CP146'

As I said, I'm very new to this, so I could be totally misunderstanding something.

**Comment by iHiD** [comment]
What is your code? (cc @exercism/python)

**Comment by floydbj** [comment]
I discovered the problem, and it was on my end. The relevant portion of code is as follows:
```python
def unique_name(self):
        """Creates a unique 5-char name of the sort LLNNN L=letter N=number"""
        """LOOP WILL BE INFINITE IF ALL AVAILABLE NAMES ARE USED!!!"""
        while True:
            characters = []
            for i in range(5):
                if i < 2:
                    characters.append(random.choice(self.letters))
                elif i < 5:
                    characters.append(random.choice(self.digits))

            name = ''.join(characters)

            if name not in self.used_names:
                self.used_names.append(name)
                return name
```
Originally, I had 'characters = []' outside the while loop. The output in that case would most likely appear normal if you're only generating a few test robots, but if you generated one having a name already in my used_names list, you'd end up with two names compressed into one string, which is of course what happened under more rigorous testing.

Test is right, noob is wrong. Let's close this. Sorry and thanks!

**Comment by iHiD** [comment]
No probs. Thanks for opening (and closing!) and enjoy Exercism :)

**Comment by yawpitch** [comment]
@floydbj a little hint to reduce complexity: look at `random.choices` and a **k** value of 2 or 3.
